docker-compose.development.yml

```diff
@@ -0,0 +1,81 @@
+version: '3.7'
+
+services:
+  flask-server:
+    container_name: flask-server
+    image: newdle
+    build:
+      context: .
+      dockerfile: Dockerfile
+    networks:
+      - default
+    volumes:
+      - './:/opt/newdle'
+      - './docker/development/newdle.cfg.docker:/opt/newdle/newdle/newdle.cfg'
+    ports:
+      - 5000:5000
+    depends_on:
+      postgres:
+        condition: service_healthy
+    environment:
+        - FLASK_RUN_HOST=0.0.0.0
+        - FLASK_ENV=development
+        - FLASK_APP=newdle.core.app
+        - NEWDLE_CONFIG=newdle.cfg
+    working_dir: /opt/newdle
+    entrypoint:
+        - bash
+        - -c
+        - |
+          pip install -q -e '.[dev]'
+          flask db upgrade
+          flask run --extra-files newdle/newdle.cfg
+    user: root
+
+  react-server:
+    container_name: react-server
+    image: newdle
+    networks:
+      - default
+    volumes:
+      - './docker/development/newdle.cfg.docker:/newdle/etc/newdle.cfg'
+      - './newdle/client:/newdle/client'
+    ports:
+      - 3000:3000
+    depends_on:
+      postgres:
+        condition: service_healthy
+      flask-server:
+        condition: service_started
+    working_dir: /newdle/client
+    environment:
+      - FLASK_URL=http://flask-server:5000
+    entrypoint:
+        - bash
+        - -c
+        - |
+            curl -sL https://deb.nodesource.com/setup_12.x | bash -
+            apt-get install -y nodejs
+            npm install
+            npm start --loglevel verbose
+    user: root
```

Comment:
I do not like the idea of running these containers as root...

Reply:
I guess it's needed to install nodejs... can we somehow avoid running the dev server itself as root afterwards?

Reply:
Not that I'm aware. But I've always used root as the standard user when developing, since this is not meant to be deployed anywhere and it's common to need to install thinks when bash the containers

Reply:
feels super dirty to me but if others do it as well so be it ;)

let's just make sure that running the dockerfile does not write any root-owned file outside the container, volumes are fine of course, but i would for example not want it to create a `node_modules` in the source folder outside the container that's owned by root